cmake/clang_tidy.cmake

```diff
@@ -3,10 +3,20 @@ option (ENABLE_CLANG_TIDY "Use clang-tidy static analyzer" OFF)
 
 if (ENABLE_CLANG_TIDY)
 
-    find_program (CLANG_TIDY_PATH NAMES "clang-tidy" "clang-tidy-15" "clang-tidy-14" "clang-tidy-13" "clang-tidy-12")
+    find_program (CLANG_TIDY_CACHE_PATH NAMES "clang-tidy-cache")
+    if (CLANG_TIDY_CACHE_PATH)
+        find_program (_CLANG_TIDY_PATH NAMES "clang-tidy" "clang-tidy-15" "clang-tidy-14" "clang-tidy-13" "clang-tidy-12")
```

Comment:
No need to prefix with _, you could just do:
```cmake
find_program (CLANG_TIDY_PATH NAMES "clang-tidy" "clang-tidy-15" "clang-tidy-14" "clang-tidy-13" "clang-tidy-12")
set (CLANG_TIDY_PATH "${CLANG_TIDY_CACHE_PATH};${CLANG_TIDY_PATH}" CACHE STRING "a doc")
```
(and pull out `find_program` so it is not replicated in both if branches.)

Reply:
I am pretty sure I've tried it, but let's see if it works. Thanks!

Reply:
I have tried it locally and the command line looked the same as before.

Reply:
see https://github.com/ClickHouse/ClickHouse/pull/42913#discussion_r1012696241